RetrofitLab/app/src/main/java/com/scottlindley/retrofitlab/MainActivity.java

```diff
@@ -0,0 +1,87 @@
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class MainActivity extends AppCompatActivity {
+    public static final String API_KEY = "8125261db99aefc2183578b967646acc";
+    private EditText mCityEdit;
+    private Button mFetchButton;
+    private TextView mCityName, mWeather, mHumidity, mPressure, mTemperature;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mCityEdit = (EditText)findViewById(R.id.city_edit);
+        mFetchButton = (Button)findViewById(R.id.search_button);
+        mCityName = (TextView)findViewById(R.id.city_name);
+        mWeather = (TextView)findViewById(R.id.weather_text);
+        mHumidity = (TextView)findViewById(R.id.humidity_text);
+        mPressure = (TextView)findViewById(R.id.pressure_text);
+        mTemperature = (TextView)findViewById(R.id.temp_text);
+
+        mFetchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getCityWeather(mCityEdit.getText().toString());
+            }
+        });
+    }
+
+    public void getCityWeather(String query){
+        ConnectivityManager connectivityManager =
+                (ConnectivityManager)getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo info = connectivityManager.getActiveNetworkInfo();
+        if(info!=null && info.isConnected()){
+            String baseUrl = "http://api.openweathermap.org";
+            Retrofit retrofit = new Retrofit.Builder()
+                    .baseUrl(baseUrl)
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .build();
+
+            OpenWeatherMapService service = retrofit.create(OpenWeatherMapService.class);
+            Call<City> call = service.getCity(query, API_KEY, "imperial");
+            call.enqueue(new Callback<City>() {
+                @Override
+                public void onResponse(Call<City> call, Response<City> response) {
+                    try{
+                        mCityName.setText(response.body().getName());
+                        mWeather.setText(response.body().getWeather().get(0).getDescription());
+                        mHumidity.setText("Humidity: "+response.body().getMain().getHumidity()+"%");
+                        double psi = Math.round(response.body().getMain().getPressure() * 001.45038);
+                        psi /= 100;
+                        mPressure.setText("Pressure: "+psi+" PSI");
+                        mTemperature.setText("Temperature: "+response.body().getMain().getTemp().toString()+"\u2109");
+                    }catch(Exception e){
+                        e.printStackTrace();
+                    }
+                }
+
+                @Override
+                public void onFailure(Call<City> call, Throwable t) {
+                    t.printStackTrace();
```

Comment:
Might want to send a quick Toast saying "unable to retrieve weather" or something